spring-jdbc/src/main/java/org/springframework/jdbc/core/metadata/TableMetaDataContext.java

```diff
@@ -265,18 +266,19 @@ public List<Object> matchInParameterValuesWithInsertColumns(SqlParameterSource p
 	 * @param inParameters the parameter names and values
 	 */
 	public List<Object> matchInParameterValuesWithInsertColumns(Map<String, ?> inParameters) {
-		List<Object> values = new ArrayList<>(inParameters.size());
+		List<Object> values = new ArrayList<>(this.tableColumns.size());
+		
+		Map<String, Object> caseInsensitiveLookup = new HashMap<>(inParameters.size());
```

Comment:
Spring itself provides a `LinkedCaseInsensitiveMap` which provides this functionality already. Wouldn't it be easier to use that, it seems to be used in other places in the JDBC access already.